src/game-engine/GameEngineA.ts

```diff
@@ -178,33 +133,24 @@ export default class GameEngineA {
     return state;
   }
 
-  receivedPreFundSetup(position: PreFundSetup) {
-    if (!(this.state instanceof State.WaitForPreFundSetupB)) { return this.state };
+  receivedPreFundSetup(position: PreFundSetupB) {
+    if (!(this.state instanceof State.WaitForPreFundSetup)) { return this.state };
 
-    const { channel, stake, balances } = this.state;
     return this.transitionTo(
-      new State.ReadyToFund({ channel, stake, balances })
+      new State.WaitForFunding({ position })
     );
   }
 
-  receivedPostFundSetup(position: PostFundSetup) {
-    if (!(this.state instanceof State.WaitForPostFundSetupB)) { return this.state };
+  receivedPostFundSetup(position: PostFundSetupB) {
+    if (!(this.state instanceof State.WaitForPostFundSetup)) { return this.state };
 
-    const { channel, stake, balances } = this.state;
+    const { stake, balances } = this.state;
 
-    if (this.state.stake > this.state.balances[0] || this.state.stake > this.state.balances[0]) {
-      return this.transitionTo(
-        new State.InsufficientFundsA({
-          channel, balances,
-        })
-      )
+    if (stake > balances[0] || stake > balances[0]) {
```

Comment:
this should be `if (stake > balances[0] || stake > balances[1]) {`
(the second `[0]` is currently incorrect)

Reply:
Nice catch.